respy/pre_processing/model_processing.py

```diff
@@ -246,11 +246,26 @@ def _sort_shocks_cov_chol(optim_paras, params, type_):
 
 
 def _parse_measurement_errors(optim_paras, params):
-    """Parse correctly sorted measurement errors."""
+    """Parse correctly sorted measurement errors.
+
+    optim_paras["is_meas_error"] is only False if there are no meas_error sds in params,
+    not if they are all zero. Otherwise we would introduce a kink into the likelihood
+    function.
+
+    """
     meas_error = np.zeros(len(optim_paras["choices"]))
 
-    labels = [f"sd_{choice}" for choice in optim_paras["choices_w_wage"]]
-    meas_error[: len(labels)] = params.loc["meas_error"].loc[labels].to_numpy()
+    if "meas_error" in params.index:
+        optim_paras["is_meas_error"] = True
+        labels = [f"sd_{choice}" for choice in optim_paras["choices_w_wage"]]
+        assert set(params.loc["meas_error"].index) == set(labels), (
```

Comment:
This is not necessary. The following ``.loc`` will fail if not all measurement errors are provided.

Does it make sense to have measurement errors in occupation a and b but not c? Then, we could set a default for missing measurement errors.

Reply:
The assert mainly is there to make sure that no measurement errors for choices without wage. Because this would not raise an error otherwise and could screw up the optimization.

Reply:
If we just discard the parameters, than the optimizer works on a useless parameter and hopefully recognize quickly, that is an hopeless case. Or the assert message should state that measurement errors should only be used for working alternatives.

Reply:
Yes, I adjusted the error message. I think it's better not to leave this to the optimizer.